### DISCUSSION TITLE
樹莓派Raspberry Pi簡介

### DISCUSSION THREAD

**Discussion by JessyCheng2003** [1101-GS4719-C/Python-Community-GS4719-C#1267] [General]
這學期修了一門選修課，常會用到樹莓派，做了很多實驗還是搞不懂它到底是什麼，只覺得聽起來很好吃..........。
所以我決定查資料更了解它也和大家分享!

### **一、What is it?**
樹莓派（Raspberry Pi），是一款基於Linux的單晶片電腦，由英國的樹莓派基金會所開發。Raspberry Pi 的最初設計是為了教育目的，以低價硬體及自由軟體促進學校的基本電腦科學教育，讓學生可以通過這個平台學習基礎的電腦科學知識和編程技能。也由於它的低價格和高度的可擴展性(搭載了各種不同的接口和插槽，可以透過這些插槽和接口來連接各種不同的其他設備，如攝像頭、麥克風、LCD 屏幕等等。)，Raspberry Pi 也成為了開發人員、研究人員和 DIY 愛好者的熱門選擇。

### 二、What is it used for?
以下是一些 Raspberry Pi 的實際應用例子：

1. 智能家居系統：我們可以使用 Raspberry Pi 搭配各種傳感器和控制器，來創建智能家居系統。例如:使用 Raspberry Pi 控制照明和空調，使用溫度傳感器和濕度傳感器來監測室內的溫度和濕度。

2. 智能花盆：使用 Raspberry Pi 搭配土壤濕度傳感器和水泵，當花盆的土壤變乾時，Raspberry Pi 會自動開啟水泵來灌溉花卉。

3. 監控系統：使用 Raspberry Pi 搭配攝像頭和運動檢測器，當運動檢測器偵測到有運動時，Raspberry Pi 會自動開啟攝像頭並拍攝照片或錄影。

4. 網絡存儲裝置：用 Raspberry Pi 搭配外接硬盤，可以通過網絡將文件傳輸到這個裝置中，然後通過任何一個連接到網絡的設備來訪問這些文件。

5. 遊戲機：使用 Raspberry Pi 搭配遊戲控制器和操作系統，可以在 Raspberry Pi 上安裝各種不同的遊戲，然後使用遊戲控制器來進行遊戲。

6. 3D打印機：使用 Raspberry Pi 搭配 3D 打印機控制板， 可以控制打印機的運動和溫度，從而實現 3D 打印。

總之，Raspberry Pi 可以應用於許多不同的領域，只要你不缺乏想像力和創造力!

**Comment by Hahahahbla** [comment]
你的介紹非常全面且詳細，Raspberry Pi 確實是一個功能強大且用途廣泛的單板電腦。

總之，Raspberry Pi 是一個用途廣泛且強大的工具，適合各種創意和項目。無論是智能家居、監控系統還是遊戲機和3D打印機，只要有創意和動手能力，都可以利用 Raspberry Pi 實現。希望這些介紹能激發你的創造力，探索更多Raspberry Pi的可能性。

**Comment by NcuJerryKao** [comment]
我也有找到一些 Raspberry Pi 的實際應用例子，歡迎各位參考：

[micro.rohm.com - 從歷史到使用方法的全面瞭解！電子作品創作不可或缺的](https://micro.rohm.com/tw/deviceplus/how-tos/raspberrypi-guide/basic-of-raspberrypi/)
[piepie.com.tw - 案例Archives](https://piepie.com.tw/tag/%E6%A1%88%E4%BE%8B)
[shumeipai.nxez.com - 树莓派行业应用案例](https://shumeipai.nxez.com/raspberry-pi-industrial-application-cases)
[speakerdeck.com - 樹莓派在物聯網上的應用案例](https://speakerdeck.com/piepie_tw/cases-study-on-raspberry-pi-for-iot)
[makerpro.cc - 不只學習和玩玩！樹莓派產業實務應用觀察](https://makerpro.cc/2022/06/observation-of-raspberry-pi-for-industrial-application/)
[tenlong.com.tw - 樹莓派<Raspberry Pi>實戰指南(手把手教你掌握100個精彩案例)](https://www.tenlong.com.tw/products/9787302401186)

**Comment by titan92601** [comment]
我認為樹莓派是一種基於Linux的單晶片電腦，由英國的樹莓派基金會開發。主要用於教育和創造，可應用於智能家居系統、監控系統、智能花盆、網絡存儲裝置、遊戲機、3D打印機等領域。具有低價格和高度可擴展性，受到開發人員、研究人員和DIY愛好者的歡迎。樹莓派的出現推動了教育和科技的普及，為學習基礎電腦科學和編程技能提供了平台。

**Comment by SusuTima** [comment]
之前在系上有用樹莓派做過自走車，不過程式碼是網路上載下來了的，也不算真正的瞭解樹莓派。後來有幸又得知了樹莓派可以放進立方衛星中，去搜集關於大氣的資料，才發現原來樹莓派真的是一個好用的東東，不只很好學，操作相對簡單，應用層面也很廣，是一個有利於學生自我鑽研的好東西。

**Comment by JustAHedgehog** [comment]
我的論點比較不同，雖然樹莓派的功能確實很強大，網路上也有不知無數的大神將樹莓派應用在各方面，例如我之前發現國外有人利用在樹莓派中搭載AI並裝載在無人機上，實現所謂的AI Drone。
然而有一點是我覺得是他至今為止的硬傷，熱當。
隨著演算力越來越強大，所造成熱損耗就越來越多，而這種PC板的焊腳又不能承受那麼高溫，一不小心就會直接整個板無法使用，尤其是4B，再加上數莓派一個就不便宜，所以在使用都強烈建議安裝風扇以及散熱鰭片去強制對流散熱他（除非你也想裝在無人機上那就忽略我說的XD）

**Comment by JustAHedgehog** [comment]
這邊再補充一個技術相關內容，目前樹莓派已經推出到第五代了，名稱叫做Raspberry Pi 5 Model B，一個$2200起跳（不含稅，沒錯購買這東西還要繳稅金喔），因為發現都沒有人放照片，所以這邊提供一下樹莓派外型。
![image](https://github.com/NCU-GS4719-Python/Python-Community/assets/143854000/0eb8edde-390a-4b9e-a8bb-836efaba2956)

硬體方面從第四代 (Raspberry Pi 4 Model B) 開始在記憶體分等級了，4B分成1GB/2GB/4GB/8GB四種版本，5B則是分成4GB/8GB兩個版本，如果要購買的話可以上**台灣樹莓派**或**飆機器人**上購買，這兩個網站是我查下來滿有公信力的網站（個人是去台灣樹莓派購買的，飆機器人則是看到系上助教是用那網站購買），最後在散熱方面這兩個網站也有提供相關套件，在此提供一張圖片提供參考：
![image](https://github.com/NCU-GS4719-Python/Python-Community/assets/143854000/d2fc0241-e261-4ffb-91d2-76c7459e4ea4)
希望能幫助想要購買樹莓派的人先評估價錢，以免走了一些冤望路

**Comment by Shang621** [comment]
樹莓派（Raspberry Pi）是一款小型單板電腦，專為教育和DIY項目而設計。它價格低廉，體積小巧，但性能強大，具有多種輸入輸出接口，如USB、HDMI和GPIO引腳，可支持各種應用場景。樹莓派適用於學習編程、製作電子項目、構建智能家居系統等，並廣泛應用於學校、家庭和工業領域。它的開放性和豐富的社區支持使得使用者可以輕鬆入門，並享受到無限創意的樂趣。

**Comment by JustAHedgehog** [comment]
在此分享Pypal的回應，他推薦「RPi.GPIO」這個Python模組，允許初學者在還沒有學習linux系統前可以使用Python程式語言來控制樹莓派的通用輸入輸出引腳，這對於製作機械工程領域的自動化系統或機器人專案非常有幫助，以下是他的回應：
如果你對樹莓派感興趣，我推薦你探索一下「RPi.GPIO」這個Python模組，它允許你使用Python程式語言來控制樹莓派的通用輸入輸出引腳，這對於製作機械工程領域的自動化系統或機器人專案非常有幫助。
